junit5-extension/src/test/java/io/kroxylicious/testing/kafka/junit5ext/ParameterExtensionTest.java

```diff
@@ -126,167 +118,177 @@ public void producerConfiguration(KafkaCluster cluster,
+            throws Exception {
         assertSameCluster(clusterA, adminA);
         assertSameCluster(clusterARef, adminA);
-        assertEquals(1, describeCluster(clusterA.getKafkaClientConfiguration()).nodes().get().size());
-        assertEquals(1, describeCluster(adminA).nodes().get().size());
-        assertSame(clusterA, clusterARef);
+        assertThat(clusterA).isSameAs(clusterARef);
+        assertClusterIdAndSize(clusterA, 1);
     }
 
     // multiple clients connected to the same cluster (e.g. different users)
     @Test
-    public void twoClusterParameterAndTwoAdmin(
-                                               @BrokerCluster(numBrokers = 1) @Name("A") KafkaCluster cluster1,
-                                               @Name("A") Admin admin1,
-                                               @Name("A") Admin admin2)
-            throws ExecutionException, InterruptedException {
-        var dc1 = describeCluster(cluster1.getKafkaClientConfiguration());
-        assertEquals(1, dc1.nodes().get().size());
+    void clusterParameterAndTwoAdmin(
+                                     @BrokerCluster(numBrokers = 1) @Name("A") KafkaCluster cluster1,
+                                     @Name("A") Admin admin1,
+                                     @Name("A") Admin admin2)
+            throws Exception {
         assertSameCluster(cluster1, admin1);
         assertSameCluster(cluster1, admin2);
-        assertNotSame(admin1, admin2);
+        assertThat(admin1).isNotSameAs(admin2);
     }
 
     @Test
-    public void zkBasedClusterParameter(@BrokerCluster @ZooKeeperCluster KafkaCluster cluster)
-            throws ExecutionException, InterruptedException {
-        var dc = describeCluster(cluster.getKafkaClientConfiguration());
-        assertEquals(1, dc.nodes().get().size());
-        assertNull(cluster.getClusterId(),
-                "KafkaCluster.getClusterId() should be null for ZK-based clusters");
+    void zkBasedClusterParameter(@BrokerCluster @ZooKeeperCluster KafkaCluster cluster)
+            throws Exception {
+        assertClusterSize(cluster, 1);
+        assertThat(cluster.getClusterId())
+                .withFailMessage("KafkaCluster.getClusterId() should be null for ZK-based clusters")
+                .isNull();
     }
 
     @Test
-    public void kraftBasedClusterParameter(@BrokerCluster @KRaftCluster KafkaCluster cluster)
-            throws ExecutionException, InterruptedException {
-        var dc = describeCluster(cluster.getKafkaClientConfiguration());
-        assertEquals(1, dc.nodes().get().size());
+    void kraftBasedClusterParameter(@BrokerCluster @KRaftCluster KafkaCluster cluster)
+            throws Exception {
+        assertClusterIdAndSize(cluster, 1);
+    }
+
+    @Test
+    void saslPlainAuth(@BrokerCluster @User(user = "alice", password = "foo") KafkaCluster cluster) {
+        doAuthExpectSucceeds(cluster, "alice", "foo");
```

Comment:
hmmm. the repetition of `alice` and `foo` gives me pause. Should we be injecting a User/principal into the test? instead of purely annotating the cluster?

Reply:
maybe, i think we can leave that discussion for another day, when there is more time to consider a deeper refactoring or even a v2.